File: character_match.py
# Don't forget to run the tests (and create some of your own)
# Part 1
def is_character_match(string1, string2):
    string1 = string1.lower().replace(' ', '')
    string2 = string2.lower().replace(' ','')
    s1 = ''.join(sorted(string1))
    s2 = ''.join(sorted(string2))
    if s1 == s2:
        return True
    return False
# Part 2
def anagrams_for(word, list_of_words):
    word = word.lower()
    s1 = ''.join(sorted(word))
    my_dict = {}
    for x in list_of_words:
        new_word = ''.join(sorted(x))
        my_dict[x] = new_word
    result = []
    for key in my_dict:
        if s1 == my_dict[key]:
            result.append(key)
    return result

File: characterMatch.js
exports.isCharacterMatch = function(string1, string2) {
    let str1= ''
    let str2= ''
    
    for (let letter of string1) {
        if (letter !== ' ') {
            str1+= letter
        }
    }

    for (let letter of string2) {
        if (letter !== ' ') {
            str2+= letter
        }
    }

    let firstString = str1.toLowerCase().split('').sort().join('')
    let secondString = str2.toLowerCase().split('').sort().join('')

    if (firstString == secondString) {
        return true
    } return false
};

exports.anagramsFor = function(word, listOfWords) {
    let matched = []
    let testWord = word.toLowerCase().split('').sort().join('')
    let obj = {}
    for (let w of listOfWords) {
        newWord = w.toLowerCase().split('').sort().join('')
        obj[w]=newWord
    }
    for (let w in obj) {
        if (obj[w] == testWord) [
            matched.push(w)
        ]
    }
    return matched

};
